pkg/macvlan/macvlan.go

```diff
@@ -270,6 +271,79 @@ func macvlanCmdDel(args *skel.CmdArgs) error {
 	return err
 }
 
+// validatePortRange validates the destination port value provided in the NAD.
+// Accepts an argument of type string and returns error if the provided port value is invalid.
+func validatePortRange(port string) error {
+	dport, err := strconv.Atoi(port)
+	if err != nil {
+		return err
+	}
+	if dport < 0 || dport > 65535 {
+		return fmt.Errorf("Port number out of range %v", port)
+	}
+	return nil
+}
+
+// generateDNATIPTablesRules creates the necessary IPTable rules to DNAT packets to remote destination.
+// Accepts an array of strings repsenting allowedDestinations to which the router can talk to.
+// Returns an error if invalid user input is detected at any point.
+func generateDNATIPTablesRules(ipt *iptables.IPTables, allowedDestinations []string) error {
+	if len(allowedDestinations) == 0 {
+		logging.Debugf("No destination information has been provided")
+		return nil
+	}
+
+	for _, allowedDestination := range allowedDestinations {
+
+		destination := strings.Split(allowedDestination, " ")
+
+		if len(destination) == 1 {
+			// should be <IPaddress/mask> format
+
+			dest, _, err := net.ParseCIDR(destination[0])
+			if err != nil {
+				logging.Errorf("Unable to parse destination IP address %v: %v", dest, err)
+				return fmt.Errorf("Unable to parse destination IP address %v: %v", dest, err)
+			}
+
+			ipt.Append("nat", "PREROUTING", "-i", "eth0", "-j", "DNAT", "--to-destination", dest.String())
+			logging.Debugf("Added iptables rule: iptables -t nat PREROUTING -i eth0 -j DNAT --to-destination %s", dest.String())
+		} else if len(destination) == 3 || len(destination) == 4 {
+			// should be <localport protocol IPaddress/mask> format
+
+			if err := validatePortRange(destination[0]); err != nil {
```

Comment:
I wonder if this plus the check on the protocol could be compacted in a single "validateDestionation" function, making this a bit more readable

Reply:
ideally, I'd prefer to create a new file `validation.go` in the future where we can move the validation stuff.